src/main/java/eu/europa/ted/eforms/sdk/component/SdkComponentFactory.java

```diff
@@ -126,22 +142,40 @@ protected <T> T getComponentImpl(String sdkVersion, final SdkComponentType compo
   }
 
   protected <T> T getComponentImpl(String sdkVersion, final SdkComponentType componentType,
-      final String qualifier, final Class<T> intf, Object... initArgs) throws InstantiationException {
+      final String qualifier, final Class<T> intf, Object... initArgs)
+      throws InstantiationException {
 
     String normalizedVersion = normalizeVersion(sdkVersion);
 
     ComponentSelector selector = new ComponentSelector(componentType, qualifier);
 
+    Map<ComponentSelector, SdkComponentDescriptor<?>> map =
+        Optional.ofNullable(componentsMap.get(normalizedVersion))
+            .orElseGet(Collections::emptyMap);
+
+    if (logger.isDebugEnabled()) {
+      logger.debug("selector componentType={}", selector.componentType);
+      logger.debug("selector qualifier={}", selector.qualifier);
+      logger.debug("normalized version={}", normalizedVersion);
+      for (Entry<ComponentSelector, SdkComponentDescriptor<?>> entry : map.entrySet()) {
+        logger.debug(
+            "entry key componentType={}, key qualifier={}, value={}",
+            entry.getKey().componentType,
+            entry.getKey().qualifier,
+            entry.getValue().getImplType().getName());
+      }
+    }
+
     @SuppressWarnings("unchecked")
     SdkComponentDescriptor<T> descriptor =
-        (SdkComponentDescriptor<T>) Optional.ofNullable(componentsMap.get(normalizedVersion))
-            .orElseGet(Collections::emptyMap).get(selector);
+        (SdkComponentDescriptor<T>) map.get(selector);
+    logger.debug("descriptor descriptor={}", descriptor);
```

Comment:
I don't think this log statement is useful, it outputs things like:
"descriptor descriptor=eu.europa.ted.eforms.sdk.component.SdkComponentDescriptor@25067551"

Reply:
OK, line removed.